R/rows-db.R

```diff
@@ -87,6 +93,7 @@ rows_update.tbl_dbi <- function(x, y, by = NULL, ...,
     invisible(x)
   } else {
     # Checking optional, can rely on primary key constraint
+    # FIXME: contrary to doc currently also checks if `in_place = FALSE`
```

Comment:
Does the documentation need to change here? We want to offer checking in both cases, and enable by default only if `in_place = FALSE`.

Reply:
Do you want checks by default if `in_place = NULL`? If not we need to tweak.

Is your point, that if someone does `in_place = TRUE`, they know what they're doing and no additional checks are needed, is that right?

Reply:
AFAICT, the implementation checks

- by default
- if `in_place = FALSE`
- if `in_place = TRUE, check = TRUE`

This is too convoluted, agree to simplify. We should check by default, with a message if it's likely that the user wants the checks off.

Reply:
That part would be consistent with dplyr, and inconsistent with `add_pk()`. I guess we can't have both.

Reply:
Ok, just to make sure I got it right:

- `check = TRUE` by default (`NULL` is no longer allowed)
- inform in case: `in_place = TRUE`.

Additionally, should we also inform in case: `in_place = NULL`?

Reply:
`check = NULL` remains the default, we inform if `check = NULL` if we think that `check = FALSE` is better.

We always inform with `in_place = NULL`?